shared/deployer/cluster.schema.yaml

```diff
@@ -175,6 +175,11 @@ properties:
             description: |
               Name of the hub. This will be used to determine
               the kubernetes namespace the hub is deployed into.
+          display_name:
```

Comment:
With the schema, you can force a display_name to be set.

I note that this schema is somewhat relaxed with regards to what is set here overall. For example, we of course require a name for the hub, but we don't declare we do.

I noted that you wrote fallback logic for display_name to rely on name - what about instead forcing a display name to be explicitly set, as well as a name, by updating this schema between line 171 and 172 to have a `required` field that sais `name`, `display_name` and other things be required?

Reply:
Welcome to the JSONSchema human parsing trials!!!

> We already have this?

Actually no, that you link is for the root path of cluster.yaml, not a requirement for fields in the items part of the `hubs` field.

Reply:
OMG @sgibson91 I THINK WE NEED VALIDATION OF OUR SCHEMA!!!! AAAAAAAAAARGH! :D

I think the required part you linked, won't do anything. You declare it for the `type: array`, but in reality we need it for the items in the array, not the array.

Reply:
Sometimes the easiest way to test is to unintentionally break. You are right.

<img width="958" alt="Screenshot 2022-03-02 at 19 43 24" src="https://user-images.githubusercontent.com/44771837/156437594-24a1563b-3a3f-4d97-aef7-9783937d3e26.png">

<img width="961" alt="Screenshot 2022-03-02 at 19 44 03" src="https://user-images.githubusercontent.com/44771837/156437735-8f259828-80ef-4de2-8860-25f271e53bd4.png">

Note: I initially tried removing the `name` key and we only raise a Python error because of this line:

https://github.com/2i2c-org/infrastructure/blob/60dc9ff22415a1984cbfd5d6e3b0171303386423/deployer/__main__.py#L242

Reply:
[bd22ab8](https://github.com/2i2c-org/infrastructure/pull/1048/commits/bd22ab8f2786aee96fa42f5c94f129980a1e269a)

Reply:
Now let's watch all the validation tests fail... 😭

Reply:
;) If they fail, they fail very gracefully though - it makes me happy to see such failures.